File: Algorithms/Sorting/Insertion Sort/insertion-sort.rb
=begin
    Insertion Sort
=end

def InsertionSort(arr)
    n = arr.length
    i = 1
    while i < n
        key = arr[i]
        j = i-1
        while j >=0 and key < arr[j]
            arr[j+1] = arr[j]
            j -= 1
        end
        arr[j+1] = key
        i += 1
    end
    print("Insertion Sort")
    print arr
end

InsertionSort([3, 1, 5, 8, 11, 10])

File: Algorithms/Sorting/Insertion Sort/insertion-sort.js
/*
 * Insertion Sort
 *
 */

function InsertionSort(arr) {
    for (let i = 1; i < arr.length; ++i) {
        let key = arr[i]
        let j = i-1
        while (j >= 0 && arr[j] > key) {
            arr[j+1] = arr[j]
            j--
        }
        arr[j+1] = key
    }
    return arr
}

console.log(InsertionSort([3,1,5,2,75,7,4,12,35,54,1,256,24,5736453,1,7]))
